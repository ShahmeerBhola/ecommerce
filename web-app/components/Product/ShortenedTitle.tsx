import React from 'react';
import Text from '../Text';

type Props = {
    title: string;
    fontWeight?: Styles.FontWeights;
    uppercase?: boolean;
    className?: string;
};

const ShortenedProductTitle: React.FunctionComponent<Props> = ({
    title,
    fontWeight = 'font-bold',
    uppercase = true,
    className,
}) => {
    return (
        <Text
            color="text-black"
            weight={fontWeight}
            className={`break-words ${className} ${uppercase ? 'uppercase' : ''}`}
        >
            {title}
        </Text>
    );
};

export default ShortenedProductTitle;
