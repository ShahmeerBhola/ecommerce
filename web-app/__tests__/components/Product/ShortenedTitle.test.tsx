import React from 'react';
import renderer from 'react-test-renderer';
import ShortenedTitle from '../../../components/Product/ShortenedTitle';

describe('FeaturedProduct component', () => {
  const title = 'My Product Title';

  test('it renders properly when title length is less than maxLength', () => {
    const tree = renderer.create(<ShortenedTitle title={title} maxLength={50} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when title length is more than maxLength', () => {
    const tree = renderer.create(<ShortenedTitle title={title} maxLength={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when fontWeight prop is provided', () => {
    const tree = renderer.create(<ShortenedTitle title={title} fontWeight="extrabold" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when uppercase prop is true', () => {
    const tree = renderer.create(<ShortenedTitle title={title} uppercase />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when uppercase prop is false', () => {
    const tree = renderer.create(<ShortenedTitle title={title} uppercase={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when className prop is provided', () => {
    const tree = renderer.create(<ShortenedTitle title={title} className="my-class-name" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
