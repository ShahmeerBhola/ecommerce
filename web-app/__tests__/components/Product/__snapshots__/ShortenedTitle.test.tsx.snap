// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedProduct component it renders properly when className prop is provided 1`] = `
<p
  className="font-secondary font-normal text-black text-base my-class-name uppercase font-bold"
>
  My Product Title
  
</p>
`;

exports[`FeaturedProduct component it renders properly when fontWeight prop is provided 1`] = `
<p
  className="font-secondary font-normal text-black text-base uppercase font-extrabold"
>
  My Product Title
  
</p>
`;

exports[`FeaturedProduct component it renders properly when title length is less than maxLength 1`] = `
<p
  className="font-secondary font-normal text-black text-base uppercase font-bold"
>
  My Product Title
  
</p>
`;

exports[`FeaturedProduct component it renders properly when title length is more than maxLength 1`] = `
<p
  className="font-secondary font-normal text-black text-base uppercase font-bold"
>
  My Pr
  ...
</p>
`;

exports[`FeaturedProduct component it renders properly when uppercase prop is false 1`] = `
<p
  className="font-secondary font-normal text-black text-base font-bold"
>
  My Product Title
  
</p>
`;

exports[`FeaturedProduct component it renders properly when uppercase prop is true 1`] = `
<p
  className="font-secondary font-normal text-black text-base uppercase font-bold"
>
  My Product Title
  
</p>
`;
